plugins/quotes.py

```diff
@@ -0,0 +1,217 @@
+			link = 'https://hastebin.com/'+str(d)
+			asyncio.create_task(client.message(data.target, link))
+		else:
+			asyncio.create_task(client.message(data.target, 'Failed to upload quotes onto pastebin.'))
+	
+	else:
+		"""if not, send each quote as a seperate message"""
+		for i in range(len(outquotes)):
+			quote = outquotes[i]
+			out = '['+str(quotenums[i])+'/'+str(len(quotelist))+']'
+			out += ' <'+quote[1]+'> '+quote[3]
+			asyncio.create_task(client.message(data.target, out))
+	
+
+
+@hook.hook('command', ['qinit'], admin=True)
+async def quoteinit(client, data):
+	"""admin only quote db table initiation hook, since trying to init
+	on every quote(even though the db functions account for that) is kinda wasteful"""
+	conn = client.bot.dbs[data.server]
+	print ('Initializing quote table in /persist/db/'+data.server+'.db...')
+	db.init_table(conn, 'quotes', quote_columns)
+	db.ccache()
+	print ('Initialization complete.')
+
+@hook.hook('command', ['q'])
+async def quotes(client, data):
+	"""main quote hook""" 
+	conn = client.bot.dbs[data.server]
+	split = data.message.split()
+	
+	tables = db.get_table_names(conn)
+	if 'quotes' not in tables:
+		asyncio.create_task(client.message(data.target, 'Quote table uninitialized. Please ask your nearest bot admin to run .qinit.'))
+	
+	"""TODO: fix weird ass edgecase with users when a nick that corresponds to a command"""
+	
+	if len(split) > 1:
+		if split[0] == 'add':
+			quote = split.copy()
+			nick = quote[1]
+			
+			"""remove the 'add', and the nickname from the message"""
+			
+			quote.pop(0)
+			quote.pop(0)
+			
+			quote = ' '.join(quote)
+			quotedata = (data.target, nick, data.nickname, quote, int(time.time()), '0',)
+			db.set_row(conn, 'quotes', quotedata)
+			asyncio.create_task(client.message(data.target, 'Quote added.'))
+			db.ccache()
+			return
+			
+		elif split[0] == 'delete':
+			nickquotes = db.get_row(conn, 'quotes', 'nick', data.nickname)
+			if int(split[1]):
+				if int(split[1]) > 0:
+					numarg = int(split[1])-1
+				elif int(split[1]) < 0:
+					numarg = int(split[1])
+				else:
+					numarg = 0
+					
+			nickquotes = remove_quotes(nickquotes)
+			if len(nickquotes) > 0:
+				try:
+					"""get quote to delete"""
+					quote = nickquotes[numarg] 
+				except:
+					if len(nickquotes) == 1:
+						asyncio.create_task(client.message(data.target, 'You only have 1 quote.'))
+					else:
+						asyncio.create_task(client.message(data.target, 'You only have '+str(len(nickquotes))+' quote(s).'))
+					return
+				
+				"""gotta do custom query since i couldn't find anything in db.py that suits my needs.
+				i'm using both the time and the quote itself because it's the quickest sure way to know that that quote is unique
+				afaik it's very unlikely that the bot is fast enough to process two or more inputs within a second
+				"""
+				
+				cur = conn.cursor()
+				cur.execute("UPDATE quotes SET deleted='1' WHERE msg=? AND time=?", (quote[3], quote[4]))
```

Comment:
I'll probably create something for this sorta stuff in db.py, since I don't want any locking issues like I've had in the past.